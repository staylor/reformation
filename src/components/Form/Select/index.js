import React, { Component } from 'react';
import { cx } from 'pretty-lights';
import { selectClass } from 'components/Form/styled';

/* eslint-disable react/prop-types */

// export type Choice = string | number | { label: string, value: string | number };
// export type Choices = Array<Choice>;
// export type Groups = Array<{ label: string, choices: Choices }>;

// type Props = {
//   className?: string | null,
//   multiple?: boolean,
//   onChange?: (value: any) => any,
//   bindRef?: (element: any) => void,
//   value?: string | number | Array<string | number> | null,
//   placeholder?: string,
//   choices?: Choices,
//   groups?: Groups,
//   children?: Node,
// };
//
// type State = {
//   value: any,
// };

const renderOption = choice => {
  if (typeof choice === 'object') {
    return (
      <option key={choice.value} value={choice.value}>
        {choice.label}
      </option>
    );
  }
  return (
    <option key={choice} value={choice}>
      {choice}
    </option>
  );
};

export default class Select extends Component {
  onChange = (e: { target: HTMLSelectElement }) => {
    let { value } = e.target;
    const multiple = Boolean(this.props.multiple);
    if (multiple) {
      value = [...e.target.selectedOptions].map(o => o.value);
    }
    if (this.props.onChange) {
      this.props.onChange(value);
    }
  };

  render() {
    const {
      className,
      placeholder,
      multiple,
      choices,
      groups,
      children,
      value,
      onChange,
      bindRef,
      ...rest
    } = this.props;

    return (
      <select
        {...rest}
        ref={bindRef}
        multiple={multiple ? Boolean(multiple) : false}
        className={cx(selectClass, className)}
        defaultValue={value}
        onChange={this.onChange}
      >
        {placeholder && (
          <option key={placeholder} value="">
            {placeholder}
          </option>
        )}
        {groups &&
          groups.map(group => (
            <optgroup key={group.label} label={group.label}>
              {group.choices.map(renderOption)}
            </optgroup>
          ))}
        {choices && choices.map(renderOption)}
        {children}
      </select>
    );
  }
}

Select.defaultProps = {
  className: null,
  multiple: false,
  onChange: () => null,
  bindRef: () => null,
  value: null,
  placeholder: null,
  choices: null,
  groups: null,
  children: null,
};
