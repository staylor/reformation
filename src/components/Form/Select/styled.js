import { css } from 'pretty-lights';
import { inputBaseClass } from 'components/Form/styled';

// eslint-disable-next-line
export const selectClass = css`
  ${inputBaseClass};
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;

  &[multiple] {
    display: block;
    height: auto;
    width: 100%;
  }
`;
