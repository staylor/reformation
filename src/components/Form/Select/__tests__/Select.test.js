import React from 'react';
import Select from 'components/Form/Select';
import { shallow } from 'enzyme';

const placeholder = '-- Select Flavor --';
const flavors = ['Mild', 'Hot', 'Fire'];
const flavorMap = [
  { label: 'Mild', value: 'mild' },
  { label: 'Hot', value: 'hot' },
  { label: 'Fire', value: 'fire' },
];
const tacoMap = [
  { label: 'Crunchy', value: 'crunchy' },
  { label: 'Soft', value: 'soft' },
];
const groups = [
  { label: 'Tacos', choices: tacoMap },
  { label: 'Saunce', choices: flavorMap },
];

describe('Select', () => {
  test('empty', () => {
    const wrapper = shallow(<Select />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state('value')).toEqual('');
  });

  test('add className', () => {
    const wrapper = shallow(<Select className="foo" />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('placeholder', () => {
    test('no choices', () => {
      const wrapper = shallow(<Select placeholder={placeholder} />);
      expect(wrapper).toMatchSnapshot();
    });

    test('choices', () => {
      const wrapper = shallow(<Select placeholder={placeholder} choices={flavors} />);
      expect(wrapper).toMatchSnapshot();
    });

    test('choices as objects', () => {
      const wrapper = shallow(<Select placeholder={placeholder} choices={flavorMap} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('multiple', () => {
    test('true', () => {
      const wrapper = shallow(<Select multiple />);
      expect(wrapper).toMatchSnapshot();
    });

    test('false', () => {
      const wrapper = shallow(<Select multiple={false} />);
      expect(wrapper).toMatchSnapshot();
    });

    test('string', () => {
      const wrapper = shallow(<Select multiple="multiple" />);
      expect(wrapper).toMatchSnapshot();
    });

    test('truthy', () => {
      const wrapper = shallow(<Select multiple="1" />);
      expect(wrapper).toMatchSnapshot();
    });

    test('falsey', () => {
      const wrapper = shallow(<Select multiple="0" />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('onChange', () => {
    test('value', () => {
      const func = jest.fn();
      const wrapper = shallow(<Select onChange={func} choices={flavorMap} />);
      const value = 'fire';
      wrapper.simulate('change', { target: { value } });
      expect(wrapper.state('value')).toEqual(value);
    });

    test('values', () => {
      const func = jest.fn();
      const wrapper = shallow(<Select multiple onChange={func} choices={flavorMap} />);
      wrapper.simulate('change', {
        target: { selectedOptions: [{ value: 'mild' }, { value: 'fire' }] },
      });
      expect(wrapper.state('value')).toEqual(['mild', 'fire']);
    });
  });

  describe('choices', () => {
    test('choices', () => {
      const wrapper = shallow(<Select choices={flavors} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.state('value')).toEqual('');
    });

    test('choices as objects', () => {
      const wrapper = shallow(<Select choices={flavorMap} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.state('value')).toEqual('');
    });

    test('choice', () => {
      const value = 'medium';
      const wrapper = shallow(<Select choices={flavorMap} value={value} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.state('value')).toEqual(value);
    });
  });

  describe('groups', () => {
    test('no value', () => {
      const wrapper = shallow(<Select groups={groups} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.state('value')).toEqual('');
    });

    test('value', () => {
      const value = 'mild';
      const wrapper = shallow(<Select groups={groups} value={value} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.state('value')).toEqual(value);
    });

    test('values', () => {
      const value = ['soft', 'mild'];
      const wrapper = shallow(<Select multiple groups={groups} value={value} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.state('value')).toEqual(value);
    });
  });

  describe('groups', () => {
    test('children', () => {
      const wrapper = shallow(
        <Select>
          <option value="meximelt">Mexi-melt</option>
        </Select>
      );
      expect(wrapper).toMatchSnapshot();
    });

    test('children with choices', () => {
      const wrapper = shallow(
        <Select choices={flavorMap}>
          <option value="meximelt">Mexi-melt</option>
        </Select>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
