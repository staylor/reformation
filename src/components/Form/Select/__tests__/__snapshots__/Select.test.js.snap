// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select add className 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0 foo"
  defaultValue={null}
  multiple={false}
  onChange={[Function]}
/>
`;

exports[`Select child nodes children 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={false}
  onChange={[Function]}
>
  <option
    value="meximelt"
  >
    Mexi-melt
  </option>
</select>
`;

exports[`Select child nodes children with choices 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={false}
  onChange={[Function]}
>
  <option
    key="mild"
    value="mild"
  >
    Mild
  </option>
  <option
    key="hot"
    value="hot"
  >
    Hot
  </option>
  <option
    key="fire"
    value="fire"
  >
    Fire
  </option>
  <option
    value="meximelt"
  >
    Mexi-melt
  </option>
</select>
`;

exports[`Select choices choice 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue="medium"
  multiple={false}
  onChange={[Function]}
>
  <option
    key="mild"
    value="mild"
  >
    Mild
  </option>
  <option
    key="hot"
    value="hot"
  >
    Hot
  </option>
  <option
    key="fire"
    value="fire"
  >
    Fire
  </option>
</select>
`;

exports[`Select choices choices 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={false}
  onChange={[Function]}
>
  <option
    key="Mild"
    value="Mild"
  >
    Mild
  </option>
  <option
    key="Hot"
    value="Hot"
  >
    Hot
  </option>
  <option
    key="Fire"
    value="Fire"
  >
    Fire
  </option>
</select>
`;

exports[`Select choices choices as objects 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={false}
  onChange={[Function]}
>
  <option
    key="mild"
    value="mild"
  >
    Mild
  </option>
  <option
    key="hot"
    value="hot"
  >
    Hot
  </option>
  <option
    key="fire"
    value="fire"
  >
    Fire
  </option>
</select>
`;

exports[`Select empty 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={false}
  onChange={[Function]}
/>
`;

exports[`Select groups no value 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={false}
  onChange={[Function]}
>
  <optgroup
    key="Tacos"
    label="Tacos"
  >
    <option
      key="crunchy"
      value="crunchy"
    >
      Crunchy
    </option>
    <option
      key="soft"
      value="soft"
    >
      Soft
    </option>
  </optgroup>
  <optgroup
    key="Saunce"
    label="Saunce"
  >
    <option
      key="mild"
      value="mild"
    >
      Mild
    </option>
    <option
      key="hot"
      value="hot"
    >
      Hot
    </option>
    <option
      key="fire"
      value="fire"
    >
      Fire
    </option>
  </optgroup>
</select>
`;

exports[`Select groups value 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue="mild"
  multiple={false}
  onChange={[Function]}
>
  <optgroup
    key="Tacos"
    label="Tacos"
  >
    <option
      key="crunchy"
      value="crunchy"
    >
      Crunchy
    </option>
    <option
      key="soft"
      value="soft"
    >
      Soft
    </option>
  </optgroup>
  <optgroup
    key="Saunce"
    label="Saunce"
  >
    <option
      key="mild"
      value="mild"
    >
      Mild
    </option>
    <option
      key="hot"
      value="hot"
    >
      Hot
    </option>
    <option
      key="fire"
      value="fire"
    >
      Fire
    </option>
  </optgroup>
</select>
`;

exports[`Select groups values 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={
    Array [
      "soft",
      "mild",
    ]
  }
  multiple={true}
  onChange={[Function]}
>
  <optgroup
    key="Tacos"
    label="Tacos"
  >
    <option
      key="crunchy"
      value="crunchy"
    >
      Crunchy
    </option>
    <option
      key="soft"
      value="soft"
    >
      Soft
    </option>
  </optgroup>
  <optgroup
    key="Saunce"
    label="Saunce"
  >
    <option
      key="mild"
      value="mild"
    >
      Mild
    </option>
    <option
      key="hot"
      value="hot"
    >
      Hot
    </option>
    <option
      key="fire"
      value="fire"
    >
      Fire
    </option>
  </optgroup>
</select>
`;

exports[`Select multiple false 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={false}
  onChange={[Function]}
/>
`;

exports[`Select multiple falsey 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={true}
  onChange={[Function]}
/>
`;

exports[`Select multiple string 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={true}
  onChange={[Function]}
/>
`;

exports[`Select multiple true 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={true}
  onChange={[Function]}
/>
`;

exports[`Select multiple truthy 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={true}
  onChange={[Function]}
/>
`;

exports[`Select placeholder choices 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={false}
  onChange={[Function]}
>
  <option
    key="-- Select Flavor --"
    value=""
  >
    -- Select Flavor --
  </option>
  <option
    key="Mild"
    value="Mild"
  >
    Mild
  </option>
  <option
    key="Hot"
    value="Hot"
  >
    Hot
  </option>
  <option
    key="Fire"
    value="Fire"
  >
    Fire
  </option>
</select>
`;

exports[`Select placeholder choices as objects 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={false}
  onChange={[Function]}
>
  <option
    key="-- Select Flavor --"
    value=""
  >
    -- Select Flavor --
  </option>
  <option
    key="mild"
    value="mild"
  >
    Mild
  </option>
  <option
    key="hot"
    value="hot"
  >
    Hot
  </option>
  <option
    key="fire"
    value="fire"
  >
    Fire
  </option>
</select>
`;

exports[`Select placeholder no choices 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  height: 28px;
  line-height: 28px;
  margin: 1px;
  max-width: 200px;
  padding: 2px;
  vertical-align: middle;
}

.lights-0[multiple] {
  display: block;
  height: auto;
  width: 100%;
}

<select
  className="lights-0"
  defaultValue={null}
  multiple={false}
  onChange={[Function]}
>
  <option
    key="-- Select Flavor --"
    value=""
  >
    -- Select Flavor --
  </option>
</select>
`;
